numpy/core/src/multiarray/methods.c

```diff
@@ -1948,7 +1951,11 @@ PyArray_Dump(PyObject *self, PyObject *file, int protocol)
     }
 
 #if defined(NPY_PY3K)
+#if PY_VERSION_HEX < 0x03080000 && PY_VERSION_HEX >= 0x03070000
+    cpick = PyImport_ImportModule("pickle5");
```

Comment:
It's not necessary to try to import `pickle5` when `protocol < 5`. In particular `data.dump(f)` should always work for Python 3.6 and 3.7 even if the `pickle5` is not installed.

Reply:
We actually need an explicit test for those cases.